src/components/Button/Button.tsx

```diff
@@ -1,55 +1,138 @@
+  readonly paddingX?: number;
+  readonly paddingY?: number;
+  readonly fontColor?: string;
+  readonly hoverFontColor?: string;
+  readonly bgColor: string;
+  readonly hoverBgColor?: string;
+  readonly hoverBgColorTransitionDuration?: number;
+  readonly borderColor?: string;
+  readonly hoverBorderColor?: string;
+  readonly focusBorderColor?: string;
+  readonly borderWidth?: number;
+  readonly borderRadius?: number;
+  readonly minButtonWidth?: number;
   readonly disabled?: boolean;
+  readonly className?: string;
+  readonly children: React.ReactNode;
+  readonly onClick?: React.MouseEventHandler<HTMLButtonElement>;
 }
 
 export const Button: React.FC<ButtonProps> = ({
+  type = 'button',
+  fontSize,
+  paddingX,
+  paddingY,
+  fontColor,
+  hoverFontColor,
+  bgColor = '#02c1b0',
+  hoverBgColor,
+  hoverBgColorTransitionDuration,
+  borderColor,
+  hoverBorderColor = '#02c1b0',
+  focusBorderColor,
+  borderWidth = 2,
+  borderRadius,
+  minButtonWidth,
+  disabled,
+  className,
   children,
   onClick,
-  bgColor = '#3EDC64',
-  color = 'white',
-  type,
-  className,
-  disabled = false,
   ...baseButtonProps
 }) => {
   return (
     <StyledButton
       type={type}
-      className={className}
+      fontSize={fontSize}
+      paddingX={paddingX}
+      paddingY={paddingY}
+      fontColor={fontColor}
+      hoverFontColor={hoverFontColor}
       bgColor={bgColor}
-      color={color}
-      onClick={onClick}
+      hoverBgColor={hoverBgColor}
+      hoverBgColorTransitionDuration={hoverBgColorTransitionDuration}
+      borderColor={borderColor}
+      hoverBorderColor={hoverBorderColor}
+      focusBorderColor={focusBorderColor}
+      borderWidth={borderWidth}
+      borderRadius={borderRadius}
+      minButtonWidth={minButtonWidth}
       disabled={disabled}
+      className={className}
+      onClick={onClick}
       {...baseButtonProps}>
       {children}
     </StyledButton>
   );
 };
 
 const StyledButton = styled.button<{
+  fontSize?: number;
+  paddingX?: number;
+  paddingY?: number;
+  fontColor?: string;
+  hoverFontColor?: string;
   bgColor: string;
-  color: string;
-  disabled: boolean;
+  hoverBgColor?: string;
+  hoverBgColorTransitionDuration?: number;
+  borderColor?: string;
+  hoverBorderColor: string;
+  focusBorderColor?: string;
+  borderWidth: number;
+  borderRadius?: number;
+  minButtonWidth?: number;
+  disabled?: boolean;
 }>`
-  font-size: 1rem;
-  color: ${({ color }) => color};
+  font-size: ${({ fontSize }) => pxToRem(fontSize ?? 16)};
+  white-space: nowrap;
+  color: ${({ fontColor }) => fontColor ?? `${colors.secondary.White}`};
   background-color: ${({ bgColor }) => bgColor};
-  text-align: center;
-  padding: 0.5rem 1.25rem;
-  border-radius: ${pxToRem(10)};
-  border: none;
-  opacity: ${({ disabled }) => (disabled ? 0.6 : 1)};
+  padding: ${({ paddingX, paddingY }) => {
+    const x = pxToRem(paddingX ?? 16);
+    const y = pxToRem(paddingY ?? 16);
+
+    return `${y} ${x}`;
```

Comment:
I don't think the default should be 1rem for both the x and y axis.  The original was 0.5rem for y-axis and 1.25rem for the x-axis.  Can we update this to match that?

Reply:
Updated!